.github/workflows/CI.yml

```diff
@@ -0,0 +1,94 @@
+name: CI
+
+on:
+  push:
+    paths-ignore:
+      - '.travis.yml'
+      - 'appveyor.yml'
+      - 'filecopy.bat'
+      - 'LICENSE*'
+      - 'README.md'
+  pull_request:
+    paths-ignore:
+      - '.travis.yml'
+      - 'appveyor.yml'
+      - 'filecopy.bat'
+      - 'LICENSE*'
+      - 'README.md'
+
+jobs:
+  build-windows:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: [Release, Debug]
+        toolset: [v141_xp, v141, v142]
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        submodules: recursive
+    - name: Prepare for build
+      run: |
+        mkdir build
+        cd build
+        cmake -T ${{ matrix.toolset }} -A Win32 ..
+    - name: Build
+      working-directory: build
+      run: cmake --build . -j $env:NUMBER_OF_PROCESSORS --config ${{ matrix.configuration }}
+    - name: Upload artifact
+      if: matrix.toolset == 'v141_xp'
+      uses: actions/upload-artifact@v1
+      with:
+        name: openag-${{ runner.os }}-${{ matrix.configuration }}
+        path: build\${{ matrix.configuration }}\client.dll
+
+  build-linux:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        configuration: [Release, Debug]
+        gxx-ver: [6, 7, 8]
```

Comment:
Nitpick: this is better called gcc-ver. Also, do they have gcc 9 and 10 in that repo?

Reply:
Ubuntu Bionic doesn't have 9 or 10.